src/web/userconfig_controller.cpp

```diff
@@ -0,0 +1,156 @@
+	}
+
+	if (SqlStmt_NumRows(stmt) <= 0) {
+		if (SQL_SUCCESS != SqlStmt_Prepare(stmt,
+				"INSERT INTO `%s` (`account_id`, `world_name`, `data`) VALUES (?, ?, ?)",
+				user_configs_table)
+			|| SQL_SUCCESS != SqlStmt_BindParam(stmt, 0, SQLDT_INT, &account_id, sizeof(account_id))
+			|| SQL_SUCCESS != SqlStmt_BindParam(stmt, 1, SQLDT_STRING, (void *)world_name, strlen(world_name))
+			|| SQL_SUCCESS != SqlStmt_BindParam(stmt, 2, SQLDT_STRING, (void *)data.c_str(), strlen(data.c_str()))
+			|| SQL_SUCCESS != SqlStmt_Execute(stmt)
+		) {
+			SqlStmt_ShowDebug(stmt);
+			SqlStmt_Free(stmt);
+			sl.unlock();
+			res.status = 400;
+			res.set_content("Error", "text/plain");
+			return;
+		}
+	} else {
+		if (SQL_SUCCESS != SqlStmt_Prepare(stmt,
+				"UPDATE `%s` SET `data` = ? WHERE (`account_id` = ? AND `world_name` = ?)",
+				user_configs_table)
+			|| SQL_SUCCESS != SqlStmt_BindParam(stmt, 0, SQLDT_STRING, (void *)data.c_str(), strlen(data.c_str()))
+			|| SQL_SUCCESS != SqlStmt_BindParam(stmt, 1, SQLDT_INT, &account_id, sizeof(account_id))
+			|| SQL_SUCCESS != SqlStmt_BindParam(stmt, 2, SQLDT_STRING, (void *)world_name, strlen(world_name))
+			|| SQL_SUCCESS != SqlStmt_Execute(stmt)
+		) {
+			SqlStmt_ShowDebug(stmt);
+			SqlStmt_Free(stmt);
+			sl.unlock();
+			res.status = 400;
+			res.set_content("Error", "text/plain");
+			return;
+		}
+	}
+
+	SqlStmt_Free(stmt);
+	sl.unlock();
+	res.set_content(data, "application/json");
+}
+
+HANDLER_FUNC(userconfig_load) {
+	if (!req.has_file("AID") || !req.has_file("WorldName")) {
+		res.status = 400;
+		res.set_content("Error", "text/plain");
+		return;
+	}
+
+	// TODO: Figure out when client sends AuthToken for this path, then add packetver check
+	// if (!isAuthorized(req)) {
+		// ShowError("Not authorized!\n");
+		// message.reply(web::http::status_codes::Forbidden);
+		// return;
+	// }
+
+	auto account_id = std::stoi(req.get_file_value("AID").content);
+	auto world_name_str = req.get_file_value("WorldName").content;
+	auto world_name = world_name_str.c_str();
+
+	SQLLock sl(WEB_SQL_LOCK);
+	sl.lock();
+	auto handle = sl.getHandle();
+	SqlStmt * stmt = SqlStmt_Malloc(handle);
+	if (SQL_SUCCESS != SqlStmt_Prepare(stmt,
+			"SELECT `data` FROM `%s` WHERE (`account_id` = ? AND `world_name` = ?) LIMIT 1",
+			user_configs_table)
+		|| SQL_SUCCESS != SqlStmt_BindParam(stmt, 0, SQLDT_INT, &account_id, sizeof(account_id))
+		|| SQL_SUCCESS != SqlStmt_BindParam(stmt, 1, SQLDT_STRING, (void *)world_name, strlen(world_name))
+		|| SQL_SUCCESS != SqlStmt_Execute(stmt)
+	) {
+		SqlStmt_ShowDebug(stmt);
+		SqlStmt_Free(stmt);
+		sl.unlock();
+		res.status = 400;
+		res.set_content("Error", "text/plain");
+		return;
+	}
+
+	if (SqlStmt_NumRows(stmt) <= 0) {
+		SqlStmt_Free(stmt);
+		ShowDebug("[AccountID: %d, World: \"%s\"] Not found in table, sending new info.\n", account_id, world_name);
+		sl.unlock();
+		res.set_content("{\"Type\": 1}", "application/json");
+		return;
+	}
+
+	char databuf[10000];
```

Comment:
Should be moved into a define and increased. Problems were reported already.
Applies to all buffers and their SQL counterparts.